src/lsptoolshost/copilot.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { CSharpExtensionId } from '../constants/csharpExtensionId';
+import { CopilotRelatedDocumentsReport, CopilotRelatedDocumentsRequest } from './roslynProtocol';
+import { RoslynLanguageServer } from './roslynLanguageServer';
+import { UriConverter } from './uriConverter';
+import { TextDocumentIdentifier } from 'vscode-languageserver-protocol';
+
+export async function registerCopilotExtension(languageServer: RoslynLanguageServer, channel: vscode.OutputChannel) {
+    const ext = vscode.extensions.getExtension('github.copilot');
+    if (!ext) {
+        channel.appendLine('GitHub Copilot extension not installed. Skipping call to `registerRelatedFilesProvider`');
+        return;
+    }
+    await ext.activate();
+    const relatedAPI = ext.exports as
+        | {
+              registerRelatedFilesProvider(
+                  providerId: { extensionId: string; languageId: string },
+                  callback: (
+                      uri: vscode.Uri
+                  ) => Promise<{ entries: vscode.Uri[]; traits?: { name: string; value: string }[] }>
+              ): void;
+          }
+        | undefined;
+    if (!relatedAPI) {
+        channel.appendLine(
+            'Incompatible GitHub Copilot extension installed. Skipping call to `registerRelatedFilesProvider`'
```

Comment:
It is possible to show a notification asking the user to upgrade to a new version if you think it would be valuable.  But I'd only do that if the extension is already installed